src/dune_rules/melange/melange_rules.ml

```diff
@@ -133,11 +152,11 @@ let js_targets_of_modules modules ~module_systems ~output =
   |> Path.Set.union_all
 ;;
 
-let js_targets_of_libs sctx libs ~module_systems ~target_dir =
+let js_targets_of_libs ~sctx ~scope ~module_systems ~target_dir libs =
   Resolve.Memo.List.concat_map module_systems ~f:(fun (_, js_ext) ->
     let open Memo.O in
     let of_lib lib =
-      let+ modules = impl_only_modules_defined_in_this_lib sctx lib in
+      let+ _, modules = impl_only_modules_defined_in_this_lib ~sctx ~scope lib in
```

Comment:
As in this case the first part of the tuple is unused, would it make sense to just return the `Modules.t` from `impl_only_modules_defined_in_this_lib` (unprocessed), and then call `modules_in_obj_dir` manually when it's needed in `setup_js_rules_libraries`? Just wondering if that work can be skipped when possible.

Reply:
Work wouldn’t be skipped. Modules.t is used to compute the Module.t list.